### PR TITLE
Add a component for handling keyboard shortcuts #60

### PR DESCRIPTION
**Pull request by damithc**

### CODE REVIEW

**Comment by damithc** [comment]
Still work in progress

**Comment by damithc** [comment]
@m133225 merge if ok

File: src/main/java/address/events/DeleteRequestEvent.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import address.model.ContactGroup;
+import address.model.Person;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Indicates a request for deleting a person
+ */
+public class DeleteRequestEvent {
```
Comment (m133225) [inline src/main/java/address/events/DeleteRequestEvent.java]: This class and `EditRequestEvent` don't seem to be used.

File: src/main/java/address/controller/PersonOverviewController.java [hunk 2]
```diff
@@ -147,10 +149,44 @@ private ContextMenu createContextMenu(){
...
         editMenuItem.setOnAction(e -> handleEditPerson());
         MenuItem deleteMenuItem = new MenuItem("Delete");
+        deleteMenuItem.setAccelerator(ShortcutsManager.SHORTCUT_PERSON_DELETE);
         deleteMenuItem.setOnAction(e -> handleDeletePerson());
         contextMenu.getItems().addAll(editMenuItem, deleteMenuItem);
         return contextMenu;
     }
+
+    @Subscribe
+    private void HandleDeleteRequestEvent(DeleteRequestEvent event) {
```
Comment (m133225) [inline src/main/java/address/controller/PersonOverviewController.java]: This and `HandleEditRequestEvent` are unused. Should be named `handleDeleteRequestEvent` (no capitalization of `H`).

File: src/main/java/address/MainApp.java [hunk 3]
```diff
@@ -61,6 +63,10 @@ protected void setupComponents() {
         storageManager = new StorageManager(modelManager);
         mainController = new MainController(this, modelManager, config);
         syncManager = new SyncManager();
+
+        shortcutsManager = new ShortcutsManager();
+        EventManager.getInstance().registerHandler(shortcutsManager);
```
Comment (m133225) [inline src/main/java/address/MainApp.java]: Should this be placed in the constructor of `ShortcutsManager` instead? To be consistent with the other components which involve event handlers.

**Comment by m133225** [comment]
Aside from minor style issues, I wonder if we should be using events for keyboard shortcuts? Rather, I think they should only be used to pass information from the model to the UI.
I guess we can discuss this later (together with #8).

File: src/main/java/address/MainApp.java (hunk 3 above)
Comment (damithc) [inline src/main/java/address/MainApp.java]: I'll move it.

File: src/main/java/address/events/DeleteRequestEvent.java (hunk 1 above)
Comment (damithc) [inline src/main/java/address/events/DeleteRequestEvent.java]: Removing

File: src/main/java/address/controller/PersonOverviewController.java (hunk 2 above)
Comment (damithc) [inline src/main/java/address/controller/PersonOverviewController.java]: Removing

**Pull request merged**